src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,307 @@
+        assertFalse(wrongName.matches(USERNAME));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createCorrectWordsEmptyOrMin2Max50")
+    void testWordEmptyOrMin2Max50_whenWordIsCorrect(String correctWord) {
+        assertTrue(correctWord.matches(WORD_EMPTY_OR_MIN_2_MAX_50));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createWrongWordsEmptyOrMin2Max50")
+    void testWordEmptyOrMin2Max50_whenWordIsWrong(String wrongWord) {
+        assertFalse(wrongWord.matches(WORD_EMPTY_OR_MIN_2_MAX_50));
+    }
+
+    private static List<String> createCorrectEmails() {
+
+        return List.of(
+                "pushkin@ukr.net",
+                "pushkin.145@gmail.com",
+                "pushkin@mail.ru",
+                "pushkin.pushkin@mail.ru",
+                "pushkin.pushkin@mail.mail.ru",
+                "1@ukr.net",
+                "1@111.ru");
+    }
+
+    private static List<String> createWrongEmails() {
+        List<String> wrongEmails = new ArrayList<>(List.of(
+                "pushkin@ukr.n",
+                ".145@gmail.com",
+                "pushkin@@mail.ru",
+                "1@111.11",
+                "pushkin.11",
+                "pushkin.1u",
+                "pushkin.u1",
+                "1@111.1u",
+                "1@111.1ru",
+                "1@111.ru1",
+                "1@ma~il.ru"));
+
+        List<Character> correctCharacters = List.of('.', '_', '+', '-');
+
+        wrongEmails.addAll(getStringsWithSpecialSymbol("pu", "shkin@ukr.net", correctCharacters));
+        wrongEmails.addAll(getStringsWithRussianOrUkrCharacter("pu", "shkin@ukr.net"));
+
+        return wrongEmails;
+    }
+
+    private static List<String> createCorrectPasswords() {
+        List<String> correctPasswords = new ArrayList<>(List.of(
+                "aA1",
+                "1aA",
+                "a1A",
+                "#aA1",
+                "aA1@",
+                "~aA1`",
+                "~`aA1`!@#$%^&*()_+-="));
+
+        correctPasswords.addAll(getStringsWithSpecialSymbol("a", "A1", Collections.emptyList()));
+
+        return correctPasswords;
+    }
+
+    private static List<String> createWrongPasswords() {
+        List<String> wrongPasswords = new ArrayList<>(List.of(
+                "",
+                "a",
+                "1",
+                "A",
+                "a1",
+                "aA",
+                "A1",
+                "aa1",
+                "a11",
+                "AA1",
+                " aA1",
+                "aA1 ",
+                " aA1 ",
+                "a A1",
+                "a`1",
+                "A`1",
+                "1#A",
+                "!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~"));
+
+        wrongPasswords.addAll(getStringsWithRussianOrUkrCharacter("a", "A1"));
+
+        return wrongPasswords;
+    }
+
+    private static List<String> createCorrectPhoneNumbers() {
+
+        return List.of(
+                "+38-050-223-32-23",
```

Comment:
You could make on test number here and using `Sting.replace('characterToReplace', 'newCharacter')` make it more easy to edit

Reply:
Thanks, I think about it